# DS256-2019--Project
Articles, checkpoints, stats, numbers and everything else........
## Articles

1.[Streaming Distributed DNA Sequence Alignment Using Apache Spark](https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=8251287)

The large amount of data generated by NextGeneration Sequencing (NGS) technology, usually in the order
of hundreds of gigabytes per experiment, has to be analyzed
quickly to generate meaningful variant results. The first step
in analyzing such data is to map those sequenced reads to
their corresponding positions in the human genome. One of
the most popular tools to do such sequence alignment is the
Burrows-Wheeler Aligner (BWA mem). One limitation of the
BWA program though is that it cannot be run on a cluster.
In this paper, we propose StreamBWA, a new framework
that allows the BWA mem program to run on a cluster in
a distributed fashion, at the same time while the input data
is being streamed into the cluster. It can process the input
data directly from a compressed file, which either lies on the
local file system or on a URL. Moreover, StreamBWA can start
combining the output files of the distributed BWA mem tasks
at the same time while these tasks are still being executed
on the cluster. Empirical evaluation shows that this streaming
distributed approach is approximately 2x faster than the nonstreaming approach. Furthermore, our streaming distributed
approach is 5x faster than other state-of-the-art solutions such
as SparkBWA. The source code of StreamBWA is publicly
available at https://github.com/HamidMushtaq/StreamBWA

2.[SparkBWA: Speeding Up the Alignment of High-Throughput DNA Sequencing Data](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0155461)

Next-generation sequencing (NGS) technologies have led to a huge amount of genomic data that need to be analyzed and interpreted. This fact has a huge impact on the DNA sequence alignment process, which nowadays requires the mapping of billions of small DNA sequences onto a reference genome. In this way, sequence alignment remains the most time-consuming stage in the sequence analysis workflow. To deal with this issue, state of the art aligners take advantage of parallelization strategies. However, the existent solutions show limited scalability and have a complex implementation. In this work we introduce SparkBWA, a new tool that exploits the capabilities of a big data technology as Spark to boost the performance of one of the most widely adopted aligner, the Burrows-Wheeler Aligner (BWA). The design of SparkBWA uses two independent software layers in such a way that no modifications to the original BWA source code are required, which assures its compatibility with any BWA version (future or legacy). SparkBWA is evaluated in different scenarios showing noticeable results in terms of performance and scalability. A comparison to other parallel BWA-based aligners validates the benefits of our approach. Finally, an intuitive and flexible API is provided to NGS professionals in order to facilitate the acceptance and adoption of the new tool. The source code of the software described in this paper is publicly available at https://github.com/citiususc/SparkBWA, with a GPL3 license

3.[SparkGA: A Spark Framework for Cost Effective, Fast and Accurate DNA Analysis at Scale](https://ce-publications.et.tudelft.nl/publications/1619_sparkga_a_spark_framework_for_cost_effective_fast_and_acc.pdf)

In recent years, the cost of NGS (Next Generation Sequencing)
technology has dramatically reduced, making it a viable method for
diagnosing genetic diseases. The large amount of data generated by
NGS technology, usually in the order of hundreds of gigabytes per
experiment, have to be analyzed quickly to generate meaningful
variant results. The GATK best practices pipeline from the Broad
Institute is one of the most popular computational pipelines for
DNA analysis. Many components of the GATK pipeline are not very
parallelizable though. In this paper, we present SparkGA, a parallel
implementation of a DNA analysis pipeline based on the big data
Apache Spark framework. This implementation is highly scalable
and capable of parallelizing computation by utilizing data-level
parallelism as well as load balancing techniques. In order to reduce
the analysis cost, SparkGA can run on nodes with as little memory
as 16GB. For whole genome sequencing experiments, we show that
the runtime can be reduced to about 1.5 hours on a 20-node cluster
with an accuracy of up to 99.9981%. Moreover, SparkGA is about
71% faster than other state-of-the-art solutions while also being
more accurate. The source code of SparkGA is publicly available at
https://github.com/HamidMushtaq/SparkGA1.git.

4.[Cluster-Based Apache Spark Implementation of the GATK DNA Analysis Pipeline](https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=7359893)

5.[Heterogeneous Hardware/Software Acceleration of the BWA-MEM DNA Alignment Algorithm ](https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=7372576)

6.[Spark Scheduler-DeepDive Talk:Data Bricks](https://databricks.com/session/apache-spark-scheduler)

7.[An intermediate data placement algorithm for load balancing in Spark computing environment](https://www.sciencedirect.com/science/article/pii/S0167739X16302126)

8.[Sparrow: Scalable Scheduling for Sub-Second Parallel Jobs](https://www2.eecs.berkeley.edu/Pubs/TechRpts/2013/EECS-2013-29.pdf)

9.[An enforcement of real time scheduling in Spark Streaming](https://ieeexplore.ieee.org/abstract/document/7393730)

10.[Official code repository for GATK versions 4 and up](https://github.com/broadinstitute/gatk)

11.[Design and Implementation of Scheduling Pool Scheduling Algorithm Based on Reuse of Jobs in Spark](https://ieeexplore.ieee.org/document/7866139)

12.[A Hard Real-time Scheduler for Spark on YARN](https://ieeexplore.ieee.org/document/8411083)
